tests/unittest/llmapi/apps/_test_disagg_serving_multi_nodes.py

```
@@ -0,0 +1,219 @@
+            if requests.get(url).status_code == 200:
+                print(f"endpoint {url} is ready")
+                return
+        except Exception:
+            pass
+
+
+def wait_for_endpoint_down(url: str, timeout: int = 300):
+    start = time.time()
+    while time.time() - start < timeout:
+        try:
+            if requests.get(url).status_code >= 100:
```

Ruff (0.12.2)

Probable use of `requests` call without timeout

(S113)